activerecord/lib/active_record/encryption/extended_deterministic_queries.rb

```diff
@@ -56,7 +57,7 @@ def process_arguments(owner, args, check_for_additional_values)
               owner.deterministic_encrypted_attributes&.each do |attribute_name|
                 attribute_name = attribute_name.to_s
                 type = owner.type_for_attribute(attribute_name)
-                if !type.previous_types.empty? && value = options[attribute_name]
+                if !type.previous_types.empty? && type.support_unencrypted_data? && value = options[attribute_name]
```

Comment:
This condition ` type.support_unencrypted_data?` should not be necessary. When `type#support_unencrypted_data?` returns false, the `type.previous_types` array won't include the "clean text scheme", so the array will be effectively empty unless there are previous schemes configured. So we shouldn't have to modify this file.